dgg-utilities.user.js

```diff
@@ -285,6 +290,53 @@ function injectScript() {
     );
   });
 
+// isUsername checks if the given element is a username element (can be usernames in messages themselves or in the list of current users)
+function isUsername(element) {
+  return element.classList.contains('user') || element.classList.contains('chat-user');
+}
+
+function doubleClickCopyListener(event) {
+  const target = event.target;
+
+  if (!isUsername(target)) {
+    return;
+  }
+
+  const username = target.text || target.textContent;
+
+  // if the chat input has some text, and the last character isn't already a space
+  if (textarea.value.length > 0 && textarea.value.charAt(textarea.value.length - 1) != ' ') {
+    textarea.value += ' ';
+  }
+
+  textarea.value += `${username} `;
+}
+
+// creating a double click to copy setting
+let doubleClickCopyGroup = document.createElement("div");
+doubleClickCopyGroup.className = "form-group checkbox";
+let doubleClickCopyLabel = document.createElement("label");
+doubleClickCopyLabel.innerHTML = "Double click username to append it to chat input";
+doubleClickCopyGroup.appendChild(doubleClickCopyLabel);
+let doubleClickCopyCheck = document.createElement("input");
+doubleClickCopyCheck.name = "doubleClickCopy";
+doubleClickCopyCheck.type = "checkbox";
+doubleClickCopyCheck.checked = doubleClickCopy;
+doubleClickCopyCheck.addEventListener("change", () => {
+  doubleClickCopy = doubleClickCopyCheck.checked;
+  window.localStorage.setItem(
+    "vyneer-util.doubleClickCopy",
+    doubleClickCopyCheck.checked
+  );
+  window.removeEventListener('dblclick', doubleClickCopyListener);
+  if (doubleClickCopy) {
+    // if a username is double clicked copy it to the chat input
+    window.addEventListener('dblclick', doubleClickCopyListener);
```

Comment:
This is only added inside the setting change event handler.  This needs to be added on start if the double click copy setting is true

Reply:
ahh, that's my bad, kinda rushed adding the settings checkbox Hhhehehehe
ill fix it! later MMMM

Reply:
i think i fixed it PepoG